src/api/app.py

```diff
@@ -603,6 +603,24 @@ async def check_quota(
     return {}
 
 
+async def generate_thread_name(query: str) -> str:
+    """
+    Generate a descriptive name for a chat thread based on the user's query.
+
+    Args:
+        query: The user's initial query in the thread
+
+    Returns:
+        A concise, descriptive name for the thread
+    """
+    try:
+        prompt = f"Generate a concise, descriptive title (max 50 chars) for a chat conversation that starts with this query:\n{query}\nReturn strictly the title only, no quotes or explanation."
+        response = HAIKU.invoke(prompt)
```

Comment:
We should switch this `.invoke()` call and all others in the code base to `.ainvoke()` to make the call async.

Reply:
Oh! good catch. Let me fix